File: DynamoDB-SDK-Examples/node.js/WorkingWithPartiQL/simple-select-statement.js
const REGION = "us-west-2";
// I am using the DynamoDB low level because the DocClient does not support executeStatement used with PartiQL
const { DynamoDB } = require("@aws-sdk/client-dynamodb");
const { unmarshall } = require("@aws-sdk/util-dynamodb");

// Create low level client
const dbclient = new DynamoDB({ region: REGION });

const executePartiQLStatement = async () => {
    // In PartiQL, the single quotes and double quotes matter. Double quotes are used to signify variable names,
    // single quotes are for string literals. Notice the info.directors where I am getting the array of directors
    // inside of the info map object.
    const params = {
        Statement: `SELECT info.directors FROM Movies WHERE "year" = 2016 AND "title" = 'The Flash'`,
    }
    return data = await dbclient.executeStatement(params);
}

executePartiQLStatement()
    .then((data) => {
            //Iterate through the Items array returned in the data variable and then unmarshall the DynamoDB JSON format to "regular" json format.
            data.Items.forEach(function (item) {
                //const doc = unmarshall(item);
                console.log(JSON.stringify(unmarshall(item), null, 2));
            });
        }
    )
    .catch((error) => console.error(JSON.stringify(error, null, 2)));

File: DynamoDB-SDK-Examples/python/WorkingWithPartiQL/simple-select-statement.py
from __future__ import print_function # Python 2/3 compatibility
import boto3, json

from botocore.exceptions import ClientError

# Create low level client
session = boto3.session.Session()
client = session.client('dynamodb')

pk = "1"
sk = "metadata"

try:
    response = client.execute_statement(
        Statement= 'SELECT * FROM ExampleTable WHERE "pk" = ? AND "sk" = ?',
        Parameters= [{"S": pk}, {"S": sk}]
    )
except ClientError as e:
    print(e.response['Error']['Message'])
else:
    print("The PartiQL query returned the following items:")
    for item in response['Items']:
        print(json.dumps(item, indent=4, sort_keys=True))